plugins/@grouparoo/mixpanel/src/lib/client/profile.ts

```diff
@@ -0,0 +1,90 @@
+import Mixpanel from "./mixpanel";
+
+export default class Profile {
+  client: Mixpanel;
+  baseUrl: string;
+
+  constructor(client: Mixpanel, baseUrl: string) {
+    this.client = client;
+    this.baseUrl = baseUrl;
+  }
+
+  async getByDistinctId(distinctId: string) {
+    const response = await this.client._request({
+      method: "POST",
+      url: `/api/2.0/engage`,
+      params: { project_id: this.client.projectId },
+      data: { distinct_id: distinctId },
+      baseURL: this.baseUrl,
+    });
+    if (response?.results?.length > 0) {
+      return response.results[0];
+    }
+    return null;
+  }
```

Comment:
It's fascinating they don't have this method in their "public" API client - I guess they assumed all user management would be though their site...

Reply:
yeah, it's a bit frustrating too.